legacy/messaging/src/main/java/org/jboss/as/messaging/MigrateOperation.java

```diff
@@ -494,12 +495,16 @@ private boolean connectionFactoryIsUsingInVMConnectors(OperationContext context,
     private void migrateDiscoveryGroup(ModelNode newAddOp, List<String> warnings) {
         // These attributes are not present in the messaging-activemq subsystem.
         // Instead a socket-binding must be used to configure the broadcast-group.
+        boolean mandatorySocketBinding = false;
         for (String property : asList(LOCAL_BIND_ADDRESS.getName(), GROUP_ADDRESS.getName(), GROUP_PORT.getName())) {
             if (newAddOp.has(property)) {
                 newAddOp.remove(property);
                 warnings.add(ROOT_LOGGER.couldNotMigrateDiscoveryGroupAttribute(property, pathAddress(newAddOp.get(OP_ADDR))));
+                mandatorySocketBinding = true;
             }
         }
+        if (mandatorySocketBinding)
+            newAddOp.get(SOCKET_BINDING.getName()).set("group-x-binding");
```

Comment:
The migrate operation will not create a new socket-binding resource, that's outside of its responsibility (which deals only with what is inside the messaging subsystem).
The resulting migrated messaging-activemq subsystem will not be valid but that's ok, we warned the user about it. He then has to decide how to make it valid again (create a socket-binding and reference it in the broadcast-group, or move to a jgroups-based configuration).